### ISSUE TITLE
TaskLogonType.Group  Error on Windows 7 professional

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#754]
I need to create a task that will from time to time on different user admin accounts .  
So, when I try using the "Principal.LogonType = TaskLogonType.Group;" on my computer with Windows 8, it works fine, but when I use this on a Windows 7 professional, it gives an error on the "LogonType".  
  
I just get a code from you to test this.  

    try
                {
                    TaskService ts = new TaskService();
                    TaskDefinition td = ts.NewTask();
                    td.RegistrationInfo.Description = &quot;Run  teste &quot;;
                    td.Settings.DisallowStartIfOnBatteries = false;
                    td.Settings.ExecutionTimeLimit = TimeSpan.Zero;
                    td.Principal.GroupId = &quot;BUILTIN\\Administrators&quot;;
                    td.Principal.LogonType = TaskLogonType.Group;
                    td.Triggers.Add(new LogonTrigger());
                    td.Actions.Add(new ExecAction(Application.StartupPath + &quot;\\teste.exe&quot;, null, null));
                    ts.RootFolder.RegisterTaskDefinition(&quot;teste&quot;, td);
    
                    MessageBox.Show(&quot;Test Ok&quot;);
                }
                catch (Exception ex)
                {
                    MessageBox.Show(&quot;ex: &quot; + ex.ToString());
                }

If I change the "TaskLogonType.Group;" to another type, it doesn't give any error, but it doesn't work, because I need to run as Administrators.  
  
The error message is:  

> COMException (0x80070534):(49,8):LogonType : in RegisterTaskDefinition in C:\Users\dahall\Documentos\VisualStudio2010\Projects\TaskService\TaskFolder.cs:line448 and line 391

Originally posted: 2017-05-30T20:51:02

**Comment by dahall** [comment]
Please try registering the task with the group information and let me know if that fixes the problem:  


    // Leave all setup code the same
    ts.RootFolder.RegisterTaskDefintion(&quot;teste&quot;, td, TaskCreation.CreateOrUpdate, &quot;BUILTIN\\Administrators&quot;, null, TaskLogonType.Group);



Originally posted: 2017-06-01T12:22:45

**Comment by dahall** [comment]
I tried that as well. It gives the same error.  


Originally posted: 2017-06-01T17:40:38

**Comment by dahall** [comment]
So, do you have any other idea? Or perhaps another version for me to try?  


Originally posted: 2017-06-05T06:38:19

**Comment by dahall** [comment]
I have just tested this while running under an account with Admin rights under Windows 7 (Version 6.1 Build 7601: SP1). I ran your exact code while built against the .NET 2.0 and 4.0 versions of the library. I repeatedly did not see your error.  
  
My guess is that it then has something to do with the user permissions under which you are running while executing on Windows 7. Make sure that account has Admin rights.  


Originally posted: 2017-06-09T17:15:25
